util/src/main/java/com/google/cloud/hadoop/util/authorization/AuthorizationHandler.java

```diff
@@ -0,0 +1,30 @@
+package com.google.cloud.hadoop.util.authorization;
+
+import java.nio.file.AccessDeniedException;
+import java.util.Map;
+
+/**
+ * All custom authorization handler implementation should implement this interface.
+ *
+ * <p>The class is created by reflection. All customized AuthorizationHandler should implement this
+ * interface and read arguments through hadoop's Configurable interface.
+ */
+public interface AuthorizationHandler {
+
+  /**
+   * This method is called before configure requests. The interface is designed for resource
+   * authorization. Only resource information (like bucket, object name) and action will be included
+   * in StorageRequestSummary.
+   *
+   * @param requestInformation The name of bucket user is trying to access.
+   * @throws AccessDeniedException Thrown when access denied.
+   */
+  void handle(StorageRequestSummary requestInformation) throws AccessDeniedException;
+
+  /**
+   * This method is called after instantiation and before calling handle.
+   *
+   * @param property Immutable map of properties.
+   */
+  void setConf(Map<String, String> property);
```

Comment:
What the purpose of this config? What properties could/should be passed to auth handler/

Reply:
It depends on user's implementation. If it need to communicate with outside server, for example, it may need to know the server's ip. I should make the comment more descriptive.